src/runtime_src/core/common/system.cpp

```diff
@@ -183,6 +183,18 @@ get_total_devices(bool is_user)
   return instance().get_total_devices(is_user);
 }
 
+std::string
+get_hostname()
+{
+  std::string hn;
+#ifdef __GNUC__
```

Comment:
Why this compiler directive.  
If it is because that windows is compiling the code, then this solution is incorrect.

Reply:
@stsoe Requested I put this in vs splitting up the implementation into a few files. I believe there is a disagreement here

Reply:
@dbenusov-xilinx @rozumx I am not willing to accept wiring up a complicated hierarchy of virtual functions to conditionally get some trivial OS specific functionality.  Please implement get_hostname() in same fashion as dlfcn.h are implemented (see core/common/dlcfn.h) by providing a trivial interface and conditionally compiled implementation.

The system class is meant to differentiate differences in platform / driver  functionality across various HAL layer implementations, it really is not supposed to provide an OS dispatch table for simple OS specific data.

Reply:
I will place the function into core/common/unistd.h. Seems to mirror what we are looking for with dlcfn.h